png2xyz/src/png2xyz.cpp

```diff
@@ -203,25 +203,22 @@ int main(int argc, char* argv[]) {
 		// Get palette and color count
 		png_get_PLTE(png_ptr, info_ptr, &palette, &num_palette);
 
-		// Check palette color count validity
-		if(num_palette != 256) {
-			std::cerr << "PNG file " << argv[arg]
-				<< " has lesser than 256 colors in palette."
-				<< std::endl;
-			png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
-			fclose(png_file);
-			return 1;
-		}
-
 		xyz_data = new unsigned char[768 + width * height];
 
 		// Create XYZ palette
```

Comment:
Suggestion: use a `memset` here to create an empty palette, before populating only available entries. Then the second `for`-loop is not needed.